clients/cobol-lsp-vscode-extension/src/services/ProcessorGroups.ts

```diff
@@ -206,50 +238,102 @@ function pathMatches(program: string, documentPath: string) {
         documentPath.toUpperCase();
 }
 
-function loadProcessorsConfig(
-  documentUri: string,
-): ProcessorConfig | undefined {
-  const documentPath = Uri.parse(documentUri).fsPath;
+const loadProcessorConfigurations = (
+  processorsJsonContent: string,
+): ProcessorConfig[] => {
+  const procCfg: ProcessorsConfig = JSON.parse(processorsJsonContent);
+  return procCfg.pgroups;
+};
+
+export function readProcessorGroupsFileContent(): string | undefined {
   const ws = SettingsUtils.getWorkspaceFoldersPath(true);
   if (ws.length < 1) {
     return undefined;
   }
   const cfgPath = path.join(ws[0], PROCESSOR_GROUP_FOLDER);
   const procCfgPath = path.join(cfgPath, PROCESSOR_GROUP_PROC);
+  if (!fs.existsSync(procCfgPath)) {
+    return undefined;
+  }
+  return fs.readFileSync(procCfgPath).toString();
+}
+
+export const readProgramConfigFileContent = (): string | undefined => {
+  const ws = SettingsUtils.getWorkspaceFoldersPath(true);
+  if (ws.length < 1) {
+    return undefined;
+  }
+  const cfgPath = path.join(ws[0], PROCESSOR_GROUP_FOLDER);
   const pgmCfgPath = path.join(cfgPath, PROCESSOR_GROUP_PGM);
-  if (!fs.existsSync(pgmCfgPath) || !fs.existsSync(procCfgPath)) {
+  if (!fs.existsSync(pgmCfgPath)) {
     return undefined;
   }
-  const procCfg: ProcessorsConfig = JSON.parse(
-    fs.readFileSync(procCfgPath).toString(),
-  );
-  const pgmCfg: ProgramsConfig = JSON.parse(
-    fs.readFileSync(pgmCfgPath).toString(),
-  );
-  const pgroup = matchProcessorGroup(pgmCfg, documentPath, ws[0]);
+  return fs.readFileSync(pgmCfgPath).toString();
+};
+
+export const loadProcessorsConfigForDocument = (
+  documentUri: string,
+  processorGroupsContent: string | undefined = readProcessorGroupsFileContent(),
+  programConfigContent: string | undefined = readProgramConfigFileContent(),
+  b4gContent: string | undefined = loadBridgeJsonContent(
+    Uri.parse(documentUri).fsPath,
+  ),
+): ProcessorConfig | undefined => {
+  const documentPath = Uri.parse(documentUri).fsPath;
+  const ws = SettingsUtils.getWorkspaceFoldersPath(true);
+  if (ws.length < 1) {
+    return undefined;
+  }
+  if (processorGroupsContent === undefined) {
+    return undefined;
+  }
+  const pgroups = loadProcessorConfigurations(processorGroupsContent);
+  const groupsConfig: ProcessorGroupConfiguration = {
+    groupMap: {},
+    procGroups: pgroups,
+  };
+  if (programConfigContent) {
+    const pgmCfg: ProgramsConfig = JSON.parse(programConfigContent);
+    pgmCfg.pgms.forEach((p) => {
+      groupsConfig.groupMap[p.program] = p.pgroup;
+    });
```

Comment:
what was the point of this change? The loading is equaly not validating the input and the map itself is traversed like an array later on.

Reply:
I'm not sure that I following, could you be more specific please? Which map are you talking about?